app/controllers/users_controller.rb

```diff
@@ -0,0 +1,43 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+
+    @user = User.new(name: params[:user][:name], password: params[:user][:password])
+    if @user.save
+      login @user  
+      redirect_to @user
+
+    else 
+
+      redirect_to signup_path
+
+    end 
+
+  end
+
+  def show
+    @user = User.find_by(id: params[:id])
```

Comment:
`[Optional]` Instead of finding the user each time you need it, it is better to create a `current_user` method to use it wherever you need that user.